tests/integration/metrics_test.go

```diff
@@ -228,3 +234,167 @@ func TestExitedContainersWithoutTTL(t *testing.T) {
+			Limit:   8000,
+		},
+		Network: biz.Network{
+			RxBytes:   3402,
+			RxDropped: 2,
+			RxErrors:  10,
+			RxPackets: 28,
+			TxBytes:   5,
+			TxDropped: 2,
+			TxErrors:  3,
+			TxPackets: 4,
+		},
+		BlkIO: biz.BlkIO{
+			TotalReadCount:  39,
+			TotalWriteCount: 89,
+			TotalReadBytes:  2387968,
+			TotalWriteBytes: 50,
+		},
+		CPU: biz.CPU{
+			CPUPercent:       0.15003804623431827,
+			KernelPercent:    1.19999999,
+			UserPercent:      100,
+			UsedCores:        2.8546433726,
+			LimitCores:       float64(runtime.NumCPU()),
+			UsedCoresPercent: float64(100) * 2.8546433726 / float64(runtime.NumCPU()),
+			ThrottlePeriods:  2384,
+			ThrottledTimeMS:  96.578349164,
+			Shares:           1024,
+		},
+		Memory: biz.Memory{
+			UsageBytes:            11620352,
+			CacheUsageBytes:       2310144,
+			RSSUsageBytes:         11620352,
+			MemLimitBytes:         104857600,
+			UsagePercent:          11.08203125,
+			KernelUsageBytes:      724992,
+			SwapUsageBytes:        1620352,
+			SwapOnlyUsageBytes:    18446744073699551616,
+			SwapLimitBytes:        209715200,
+			SwapLimitUsagePercent: 0.77264404296875,
+			SoftLimitBytes:        262144000,
+		},
+		RestartCount: 2,
+	}
+
+	// Create a tempDir that will be the root of our mocked filsystem
+	hostRoot := t.TempDir()
+
+	// Create our mocked container cgroups filesystem in the tempDir directory
+	// that will be a symLink to our cgroups testdata
+	cgroupsFolder := filepath.Join(hostRoot, "cgroup")
+	mockFilesystem, err := filepath.Abs(filepath.Join(relativePathToTestdataFileystem, "cgroup"))
+	require.NoError(t, err)
+
+	err = os.Symlink(mockFilesystem, cgroupsFolder)
+	require.NoError(t, err)
+
+	// Create mocked directory proc
+	err = os.Mkdir(filepath.Join(hostRoot, "proc"), 0755)
+	require.NoError(t, err)
+
+	err = generateMockedProcMountsFile(cgroupsFolder, hostRoot)
+	require.NoError(t, err)
+
+	err = generateMockedProcPIDCGroupFile(hostRoot)
+	require.NoError(t, err)
+
+	err = genereateMockedProcNetDevFile(hostRoot)
+	require.NoError(t, err)
+
+	// CgroupsFetcherMock is the raw CgroupsFetcher with mocked cpu.systemUsage and time
+	// The hostRoot is our mocked filesystem
+	cgroupFetcher, err := NewCgroupsFetcherMock(hostRoot, "cgroupfs", "", mockedTimeForAllMetricsTest, 19026130000000)
+	require.NoError(t, err)
+
+	storer := generateInMemoryStorerWithPreviousCPUState()
+	inspector := NewInspectorMock(InspectorContainerID, InspectorPID, 2)
+
+	t.Run("Given a mockedFilesystem and previous CPU state Then processed metrics are as expected", func(t *testing.T) {
+		metrics := biz.NewProcessor(storer, cgroupFetcher, inspector, 0)
+
+		sample, err := metrics.Process(InspectorContainerID)
+		require.NoError(t, err)
+		assert.Equal(t, expectedSample, sample)
+	})
+}
+
+// generateInMemoryStorerWithPreviousCPUState creates a storere with a previous CPU state
+// in order to make the processor calculate a Cpu Delta
+func generateInMemoryStorerWithPreviousCPUState() persist.Storer {
```

Comment:
nit: i think the generate prefix could be removed from the name

Reply:
done